aspnet/security/cross-site-scripting.rst

```diff
@@ -1,8 +1,256 @@
-.. include:: /../common/stub-topic.txt
+.. _security-cross-site-scripting:
 
-|stub-icon| Preventing Cross-Site Scripting
-===========================================
+Preventing Cross-Site Scripting
+===============================
 
-.. include:: /../common/stub-notice.txt
+Cross-Site Scripting (XSS) is a security vulnerability which enables an attacker to place client side scripts (usually JavaScript) into web pages.
+When other users load affected pages the attackers scripts will run, enabling the attacher to steal cookies and session tokens, change the contents
+of the web page via DOM manipulation or redirect the browser to another page. XSS vulnerabilities generally occur when an application takes user 
+input and outputs it in a page without validating or encoding or escaping it.
 
-.. _issue: https://github.com/aspnet/Docs/issues/90
\ No newline at end of file
+.. contents:: Sections:
+  :local:
+  :depth: 1
+
+Protecting your application against XSS
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+At a basic level XSS works by tricking your application into inserting a ``<script>`` tag into your rendered page, or by inserting an ``On*`` event 
+into an element. Developers should use the following prevention steps to avoid introducing XSS into their application.
+
+1. Never put untrusted data into your HTML input, unless you follow the rest of the steps below. 
+   Untrusted data is any data that may be controlled by an attacker, HTML form inputs, query strings,
```

Comment:
RO sentence. Try this:
Untrusted data is any data that may be controlled by an attacker: HTML form inputs, query strings, HTML headers, even data sourced from a database. An attacker may be able to breach your database if not your application.

I'm not exactly sure what "if not your application." means or adds to this.